.ci/scripts/check_code_format.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash -eEx
+set -o pipefail
+
+TMP_DIR="/tmp/$(basename "$0" .sh)_$$"
+rm -rf "${TMP_DIR}"
+mkdir -p "${TMP_DIR}"
+
+UCC_SRC_DIR="/opt/nvidia/src/ucc"
+cd "${UCC_SRC_DIR}"
+
+git log -1 HEAD
+git log -1 HEAD^
```

Comment:
@artemry-nv will this check only top/HEAD commit from the PR or all the commits (if PR contains multiple commits) ?

Reply:
It looks like there's no easy way to detect baseline commit in pure git (like ` ${{ github.event.pull_request.base.sha }}` in github actions), but CI will be run for each commit within a PR, so all the changes will be checked.

Reply:
I'm not sure. If the PR has, say, 3 commits and last one formats everything as desired but 1st and 2nd produce non-zero clang-diff - will this make whole PR bad?

Reply:
Anyway currently code format checker is non-critical for PR status (CI won't fail if code format isn't OK and PR merge won't be blocked) - as far as I understand this's agreed within UCC team (mentioned by @Sergei-Lebedev ). So because this it's up to the PR author to check that code formatting is OK. I'm adding log publishing through PR comments - it will look like here https://github.com/Mellanox/nccl-rdma-sharp-plugins/pull/54#issuecomment-693966544 and you can check status for each step (build, tests, different checkers, etc.) with logs.

Reply:
@vspetrov May be we should squash and merge? That way, it is always checked. We don't follow this policy (squash and merge) consistently but we can make this more consistent.

Reply:
well, this is not directly related to this PR , since indeed we only need to check latest commit in the PR (maybe last commit restores code style). But in general this is smth to discuss. Honestly, i don't know what are the rules of thumb for merge policy. I personally tend to squash (unless just forget about it). We can restrict this.